BlazarService/src/main/java/com/hubspot/blazar/util/BlazarSlackClient.java

```diff
@@ -0,0 +1,131 @@
+import com.ullink.slack.simpleslackapi.SlackSession;
+import com.ullink.slack.simpleslackapi.SlackUser;
+import com.ullink.slack.simpleslackapi.replies.ParsedSlackReply;
+import com.ullink.slack.simpleslackapi.replies.SlackMessageReply;
+
+/**
+ * This class handles all the details of sending a slack message, and verifying that it did send.
+ */
+@Singleton
+public class BlazarSlackClient {
+  private static final Logger LOG = LoggerFactory.getLogger(BlazarSlackClient.class);
+
+  private SlackSession session;
+  private MetricRegistry metricRegistry;
+
+  @Inject
+  public BlazarSlackClient(SlackSession session, MetricRegistry metricRegistry) {
+    this.session = session;
+    this.metricRegistry = metricRegistry;
+  }
+
+  public void sendMessageToChannel(String channelName, String message, SlackAttachment attachment) {
+    Optional<SlackChannel> slackChannel = Optional.fromNullable(session.findChannelByName(channelName));
+    if (!slackChannel.isPresent()) {
+      LOG.warn("No slack channel found for name {}", channelName);
+      return;
+    }
+
+    sendMessageToChannel(slackChannel.get(), message, attachment);
+  }
+
+  public void sendMessageToChannel(SlackChannel channel, String message, SlackAttachment attachment) {
+    try {
+      makeSlackMessageSendingRetryer().call(() -> sendMessage(channel, message, attachment));
+      metricRegistry.counter("successful-slack-channel-sends").inc();
+      // Here we swallow exceptions that might be thrown by our retryer or #sendMessage().
+      // We don't catch and retry any exceptions because the slack session doesn't throw us any, (it swallows them and returns null)
+    } catch (Exception e) {
+      metricRegistry.counter("failed-slack-channel-sends").inc();
+      LOG.error("Could not send slack message {}", attachment, e);
+    }
+  }
+
+  public void sendMessageToUser(String email, String message, SlackAttachment attachment) {
+    Optional<SlackUser> user = Optional.fromNullable(session.findUserByEmail(email));
+    if (!user.isPresent()) {
+      LOG.warn("Could not find user with email {} ", email);
+      return;
+    }
+    sendMessageToUser(user.get(), message, attachment);
+  }
+
+  public void sendMessageToUser(SlackUser user, String message, SlackAttachment attachment) {
+    try {
+      makeSlackMessageSendingRetryer().call(() -> sendMessage(user, message, attachment));
+      metricRegistry.counter("successful-slack-dm-sends").inc();
+      // Here we swallow exceptions that might be thrown by our retryer or #sendMessage().
+      // We don't catch and retry any exceptions because the slack session doesn't throw us any, (it swallows them and returns null)
+    } catch (Exception e) {
+      LOG.error("Could not send slack message {}", attachment, e);
+      metricRegistry.counter("failed-slack-dm-sends").inc();
+    }
+  }
+
+  private boolean sendMessage(SlackUser user, String message, SlackAttachment attachment) {
```

Comment:
Per my comment above about the session that can be injected not connected, check if session is connected before you try to send